pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -347,7 +344,7 @@ func getKubernetesExtensionClientsets() (azKubeExtensionClientset versioned.Inte
 	return azKubeExtensionClientset, nil
 }
 
-func getKubernetesClientset() (*kubernetes.Clientset, error) {
+func getKubernetesClientset() (kubernetes.Interface, error) {
```

Comment:
In Go, it is generally preferred to return (a pointer to) the concrete type rather than an interface. Is there a reason for this change?

Reply:
I got in the habit of using interface when possible as it felt to me that it broadens the usability of the function. I didn't know that I was going exactly against the convention and guided preference (slap in the head).

Reply:
General guidance is to take accept parameters but return concrete types. If a function can return multiple concrete types, then it would need to return an interface, but this is not the case here.